File: assets/composition/walking-skeleton/go-port-fake/templates-modulith/port/clock.go
// Package clock holds the Clock port: the secondary port for reading
// the current instant.
//
// It sits under internal/platform/ rather than inside a bounded
// context because no context owns it — every context needs a clock,
// and none of them defines what time means. Adapters, real and fake,
// implement it beside this package.
package clock

import "time"

// Clock is the secondary port for reading the current instant.
type Clock interface {
	// Now returns the current instant.
	Now() time.Time
}

File: assets/composition/walking-skeleton/rust-port-fake/templates/port/clock.rs
//! The `Clock` secondary port. The domain owns the trait (ports live
//! with their consumer); real and fake adapters implement it under
//! `infra`.

use std::time::SystemTime;

/// Clock is the secondary port for reading the current instant.
pub trait Clock {
    /// Returns the current instant.
    fn now(&self) -> SystemTime;
}
